otter/grade/__init__.py

```diff
@@ -155,6 +157,17 @@ def main(
     # write to CSV file
     output_df.to_csv(os.path.join(output_dir, "final_grades.csv"), index=False)
 
+    # write score summaries to files
+    if summaries:
+        grading_summary_path = os.path.join(output_dir, "grading-summaries")
+        if not os.path.exists(grading_summary_path):
+            os.mkdir(grading_summary_path)
+        for df in grade_dfs:
+            df_dict = df.to_dict()
+            if df_dict['file'][0] != POINTS_POSSIBLE_LABEL:
+                with open(os.path.join(grading_summary_path, f"{df_dict['file'][0]}.txt"), mode="w") as f:
```

Comment:
What is this writing? I thought the goal was to save the GradingResults object, which would require using `dill` to serialize it and opening the file in `wb+` mode.

Reply:
Here is an example of what is written during the tests; the notebook name is fails6H.txt
[fails6H.txt](https://github.com/user-attachments/files/16701149/fails6H.txt)

Reply:
Ah, I guess I misunderstood the goal. I thought you wanted to save the `GradingResults` object itself, not the results summary. This looks OK then.